src/Symfony/Component/VarDumper/Caster/ResourceCaster.php

```diff
@@ -19,16 +19,31 @@
  * @author Nicolas Grekas <p@tchwork.com>
  *
  * @final
+ *
+ * @internal since Symfony 7.3
  */
 class ResourceCaster
 {
+    /**
+     * @deprecated since Symfony 7.3
+     */
     public static function castCurl(\CurlHandle $h, array $a, Stub $stub, bool $isNested): array
     {
-        return curl_getinfo($h);
+        trigger_deprecation('symfony/var-dumper', '7.3', 'The "%s()" method is deprecated without replacement.', __METHOD__, CurlCaster::class);
+
+        return CurlCaster::castCurl($h, $a, $stub, $isNested);
     }
 
-    public static function castDba($dba, array $a, Stub $stub, bool $isNested): array
+    /**
+     * @param resource|\Dba\Connection $dba
+     */
+    public static function castDba(mixed $dba, array $a, Stub $stub, bool $isNested): array
     {
+        if (\PHP_VERSION_ID < 80402 && !\is_resource($dba)) {
+            // @see https://github.com/php/php-src/issues/16990
+            return $a;
+        }
+
         $list = dba_list();
         $a['file'] = $list[(int) $dba];
```

Comment:
do we have a test case that proves this works? casting a dba connection to an int matches the indexes returned by dba_list in PHP 8.4.2?

Reply:
Improved `ResourceCasterTest.php` to cover each case